doc/smart-switch/pmon/smartswitch-pmon.md

```diff
@@ -0,0 +1,806 @@
+# SmartSwitch PMON High Level Design
+
+| Rev | Date | Author | Change Description |
+| --- | ---- | ------ | ------------------ |
+| 0.1 | 12/02/2023 | Ramesh Raghupathy | Initial version|
+| 0.2 | 01/08/2024 | Ramesh Raghupathy | Updated API, CPI sections and addressed review comments |
+| 0.3 | 02/26/2024 | Ramesh Raghupathy | Addressed review comments |
+
+## Definitions / Abbreviations
+
+| Term | Meaning |
+| --- | ---- |
+| PMON | Platform Monitor |
+| DLM | Device Lifecycle Manager |
+| NPU | Network Processing Unit |
+| DPU | Data Processing Unit |
+| PDK | Platform Development Kit |
+| SAI | Switch Abstraction Interface |
+| GPIO | General Purpose Input Output |
+| PSU | Power Supply Unit |
+| I2C | Inter-integrated Circuit communication protocol |
+| SysFS | Virtual File System provided by the Linux Kernel |
+| CP | Control Plane |
+| DP | Data Plane |
+
+## 1. Introduction
+SmartSwitch offloads the Packet Processors (NPUs) and the host CPUs, freeing up resources for application performance, thereby performing layer four to layer seven functions in a cost effective and space saving way. 
+
+The specialized DPUs when built into a regular switch, can provide such a capability, which is being referred as SmartSwitch.
+Platform monitor PMON in SONiC is a container responsible for chassis management functions to ensure proper operation of the devices and peripherals in the chassis there by ensuring the proper operation of the product. 
+
+The typical lifecycle of a product involves the following stages.
+
+<p align="center"><img src="./images/lifecycle.svg"></p>
+
+The following sub-tasks are performed under each stage
+* Onboarding
+    * Boot, Shutdown, Power Cycle
+    * Rest, PCIe-Reset
+* Monitoring
+    * Device State (dpu_state)
+    * Sensors, PSUs, Cooling Devices, Thermal management
+    * Show CLIs
+* Detection and Debugging
+    * DPU Health
+    * Alarms, Syslog
+    * Console
+* RMA
+    * Inventory
+
+The purpose of this document is to provide a framework to share the state, health, alarms of the DPUs, manage the DPUs by providing support to monitor, gracefully shutdown, restart them and the associated peripherals such as thermal sensors, cooling devices, LEDs, etc.
+
+The picture below highlights the PMON vertical and its association with other logics within the SONiC architecture.
+
+<p align="center"><img src="./images/pmon-vertical.svg"></p>
+
+## 2.	Requirements and Assumptions
+### 2.1.    Onboarding
+* The SmartSwitch host PMON should be able to Power Cycle, Shutdown, Reset, and rest the PCIe link per DPU or the entire system
```

Comment:
We should have a config DB table that will represent an admin state of the DPU. For example, if we want to use a smart switch as a switch only we should have the ability to power off the DPUs and have this as a permanent configuration. This can reduce the power consumption of the box.

I see that there is a `set_admin_state(...)` API defined. But without having the configuration in the config DB any changes to the DPU admin state will have a temporary effect.

Reply:
As per the requirement the smartswitch will boot up as a switch by default, meaning that all the DPUs will remain in power down state until unless explicitly booted.  Explained this in detail. Don't see a need for config as this is the default behavior.  Let us keep it simple.

Reply:
With the proposed design there is no possibility to make DPU admin state permanent. If the DPU by default is in the down state during the system start who is responsible for bringing it up? If the user sets the DPU to the up state and then does the power cycle of the smart switch what will be the state of the DPU after the smart switch boot?

The assumption that a smart switch boots as a switch by default is based on the behavior of one specific platform. The behavior might be different on the other platforms. That is why we need to have a DPU state in the config DB.